### ISSUE TITLE
Is there a way to have FE+Extensions/BE in separate repos?

### ISSUE THREAD

**Issue by delyan-peev** [Shopify/cli#293]
Hey there,

we are currently trying to set our project up in a way that has the frontend (+extension) and backend in two separate repos. It used to work just fine with CLI 2, however, we cannot seem to figure out a way to push our theme app extension via the new CLI.

Is there a way to do it or is it compulsory that we use the specific directory structure with both FE and BE in one repo and serving the app with shopify CLI & ngrok? This is kind of a struggle for us, because we have some restrictions imposed that restrict us from using ngrok and what we do currently for development is serving our app over https at a public IP address.

I would also like to have the ability to scale the BE horizontally independently of the FE and I cannot think of a way to do it if the project(s) structure is the one advised in the documentation.

Thanks for your cooperation in advance.

Regards,
Delyan

**Comment by alvaro-shopify** [comment]
HI Delyan!, First of all thank you for reporting this question. I will try to give you as much information as possible or otherwise  I will discuss some of this points with the team to give you a possible solution.

The using of ngrok is not mandatory, although it is for now the default behaviour when using the `dev` command. To bypass it you can use the flag `--tunnel-url` to use a custom tunnel of your own.

Regarding the separation of the FE + BE I'm going to catch up some information from the team and I will come back to you with it as soon as possible

**Comment by delyan-peev** [comment]
Hey guys,

thanks for the answers. By digging through the Shopify CLI codebase a bit, I managed to achieve what I needed.

>but there's no "deployment" of your app to Shopify when you run `shopify app deploy`

That's exactly my finding, so I just kept the `/web/frontend/<FE-files>` for the frontend project and `/extensions/<extensions>` with the relevant `toml` files in these directories and the `shopify.app.toml` in the root of the project, so the CLI does not complain and the setup seems to work as expected.

I thought `shopify app deploy` *must* be used to deploy the whole app as I noticed the docs mentioned that still only Heroku is available. Anyway, thanks for the help, I got what I needed.

P.S. I think the CLI might use some better `-h` description.

Closing this.

Regards,
Delyan
